net/nettest/listener.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package nettest
+
+import (
+	"context"
+	"net"
+	"sync"
+)
+
+const (
+	bufferSize = 256 * 1024
+)
+
+// Listener is a net.Listener using Conn.
```

Comment:
but Conn's just an interface. I'd say "... using NewConn to create pairs of network connections connected in memory using a buffered pipe".